app/uk/gov/hmrc/economiccrimelevyregistration/viewmodels/checkAnswers/ContactAddressSummary.scala

```diff
@@ -30,11 +32,17 @@ object ContactAddressSummary {
       SummaryListRowViewModel(
         key = Key("checkYourAnswers.address.label"),
         value = ValueViewModel(AddressViewModel.htmlContent(answer)),
-        actions = Seq(
-          ActionItemViewModel("site.change", "#TODO").withVisuallyHiddenText(
-            messages("checkYourAnswers.address.label")
-          )
-        )
+        actions = request.registration.useRegisteredOfficeAddressAsContactAddress match {
+          case Some(true)  => Seq.empty
+          case Some(false) =>
+            Seq(
+              ActionItemViewModel("site.change", routes.IsUkAddressController.onPageLoad(CheckMode).url)
+                .withVisuallyHiddenText(
+                  messages("checkYourAnswers.address.label")
+                )
+            )
+          case _           => throw new IllegalStateException("Use registered office address as contact address answer not found")
```

Comment:
rather than throwing an exception here - can we not map over it and return the row optionally instead, otherwise we are leaking backend validation logic into our check your answer row

Reply:
here's what I mean:

```
    request.registration.useRegisteredOfficeAddressAsContactAddress.flatMap { useRegisteredOfficeAddressAsContactAddress =>
      request.registration.contactAddress.map { answer =>
        SummaryListRowViewModel(
          key = Key("checkYourAnswers.address.label"),
          value = ValueViewModel(AddressViewModel.htmlContent(answer)),
          actions = if (useRegisteredOfficeAddressAsContactAddress) { Seq.empty }
          else {
            Seq(
              ActionItemViewModel("site.change", routes.IsUkAddressController.onPageLoad(CheckMode).url)
                .withVisuallyHiddenText(
                  messages("checkYourAnswers.address.label")
                )
            )
          }
        )
      }
    }
```